store/store.go

```diff
@@ -1806,7 +1807,8 @@ func (s *Store) isStaleRead(freshness int64) bool {
 	if freshness == 0 || s.raft.State() == raft.Leader {
 		return false
 	}
-	return time.Since(s.raft.LastContact()).Nanoseconds() > freshness
+	return time.Since(s.raft.LastContact()).Nanoseconds() > freshness &&
+		s.raft.CommitIndex() == s.DBAppliedIndex()
```

Comment:
Having `s.raft.CommitIndex() == s.DBAppliedIndex()` makes the condition too strict, especially for large updates that can still finish within the allowed freshness.

Could we instead add a`Time` field in `Store` that gets updated to `LastContact` when there is no new index, and when there is, to whenever the index gets applied?

Reply:
I noticed `s.DBAppliedIndex()` is protected by a mutex. Would it make sense to replace it with a https://pkg.go.dev/sync/atomic#Uint64?

Reply:
Yeah, you're the second person to suggest this. @jtackaberry did too.

Makes sense, but I'll do that in a separate PR.